prow/userdashboard/userdashboard.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package userdashboard
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/shurcooL/githubql"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+)
+
+const githubEndpoint = "https://api.github.com"
+
+type githubClient interface {
+	Query(context.Context, interface{}, map[string]interface{}) error
+}
+
+// PullRequestQueryHandler defines an interface that query handlers should implement.
+type PullRequestQueryHandler interface {
+	Query(context.Context, githubClient) ([]PullRequest, error)
+}
+
+// UserData represents data returned to client request to the endpoint. It has a flag that indicates
+// whether the user has logged in his github or not and list of open pull requests owned by the
+// user.
+type UserData struct {
+	Login        bool
+	PullRequests []PullRequest
+}
+
+// Dashboard Agent is responsible for handling request to /user-dashboard endpoint. It will serve
+// list of open pull requests owned by the user.
+type DashboardAgent struct {
+	goac *config.GithubOAuthConfig
+
+	log *logrus.Entry
+}
+
+type Label struct {
+	ID   githubql.ID
+	Name githubql.String
+}
+
+type PullRequest struct {
```

Comment:
There are some additional fields we will probably need in this struct. The headRefOID, and milestones for example. We can probably just add fields as needed though.

Reply:
done